### ISSUE TITLE
resolve() cannot omit the value.

### ISSUE THREAD

**Issue by shtse8** [guzzle/promises#76]
the thenable cannot omit the value for resolve().

```
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function GuzzleHttp\Promise\Promise::resolve(), 0 passed
```

but it's very useful for just resolve the promise without returning the value.

**Comment by kelunik** [comment]
You could just explicitly resolve it with `null`, not too much of an issue, no?

**Comment by shtse8** [comment]
yes, but I have to input null for that argument. but its common for other Promise makers design that just resolving with null by default. In most cases, we just want to resolve the promise as the job is completed. but no value needs to be returned.

**Comment by Tobion** [comment]
Chaning this requires an interface change (PromiseInterface) and a BC break. So it need to wait until the next major version.

**Comment by GrahamCampbell** [comment]
This has been implemented in 3.0.
